src/overlays/ovl_i1/E05620.c

```diff
@@ -1,7 +1,59 @@
-#include "common.h"
+#include "global.h"
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/E05620/func_i1_801988E0.s")
+void func_i1_801988E0(void) {
+    if (D_80177E80 != 0) {
+        RCP_SetupDL(&gMasterDisp, 0x53);
+        gDPSetPrimColor(gMasterDisp++, 0, 0, 255, 255, 255, 255);
+        func_800869A0(250.0f, 50.0f, D_80177E80, 1.0f, 0, 999);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/E05620/func_i1_80198968.s")
+void func_i1_80198968(Item* item) {
+    f32 var_fv0;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/E05620/D_i1_8019B640.s")
+    switch (item->unk_46) {
+        case 0:
+            item->obj.rot.z += 1.0f;
+            Math_SmoothStepToF(&item->scale, 1.0f, 1.0f, 0.05f, 0.0f);
+            if ((item->unk_44 == 0) && (gPlayer->unk_4DC == 0)) {
+                if (item->obj.rot.y == 0.0f) {
+                    var_fv0 = 550.0f;
+                } else {
+                    var_fv0 = 1000.0f;
+                }
+                if (item->obj.pos.z > (gPlayer->unk_138 + var_fv0)) {
+                    item->unk_44 = 1;
+                    PRINTF("♪:リング未通過音\n"); // Ring not passed sound
+                    Audio_PlaySfx(0x4900402F, &D_800C5D28, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                    D_80177E80 = 0;
+                }
+            }
+            if (item->unk_4C != 0) {
+                item->unk_46 = 1;
+                item->timer_48 = 50;
+                item->info.unk_10 = 10000.0f;
+                PRINTF("♪:リング通過音\n"); // Ring passing sound
+                Audio_PlaySfx(0x49008025, &D_800C5D28, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                D_80177E80++;
+                if ((item->obj.rot.x != 0.0f) && (D_80177E80 >= 100)) {
+                    func_800BA808(gMsg_ID_20330, RCID_ROB64);
+                    return;
+                }
+            }
+            return;
```

Comment:
can this be break?

Reply:
Yeah, woops. I forgot to do that after M2C made it a return. Fixed now.